backend/src/apiserver/client_manager/client_manager.go

```diff
@@ -310,52 +347,74 @@ func initDBClient(initConnectionTimeout time.Duration) *storage.DB {
 	}
 
-	// If the old unique index idx_pipeline_version_uuid_name on pipeline_versions exists, remove it.
-	rows, err := db.Raw(`show index from pipeline_versions where Key_name='idx_pipeline_version_uuid_name'`).Rows()
-	if err != nil {
-		glog.Fatalf("Failed to query pipeline_version table's indices. Error: %s", err)
-	}
-	if err := rows.Err(); err != nil {
-		glog.Fatalf("Failed to query pipeline_version table's indices. Error: %s", err)
-	}
-	if rows.Next() {
-		db.Exec(`drop index idx_pipeline_version_uuid_name on pipeline_versions`)
+	// Because PostgreSQL was supported later, there's no need to delete the relic index
+	if driverName == "mysql" {
+		// If the old unique index idx_pipeline_version_uuid_name on pipeline_versions exists, remove it.
+		rows, err := db.Raw(`show index from pipeline_versions where Key_name='idx_pipeline_version_uuid_name'`).Rows()
+		if err != nil {
+			glog.Fatalf("Failed to query pipeline_version table's indices. Error: %s", err)
+		}
+		if err := rows.Err(); err != nil {
+			glog.Fatalf("Failed to query pipeline_version table's indices. Error: %s", err)
+		}
+		if rows.Next() {
+			db.Exec(`drop index idx_pipeline_version_uuid_name on pipeline_versions`)
+		}
+		defer rows.Close()
 	}
-	defer rows.Close()
 
 	return storage.NewDB(db.DB(), storage.NewMySQLDialect())
 }
 
-// Initialize the connection string for connecting to Mysql database
-// Format would be something like root@tcp(ip:port)/dbname?charset=utf8&loc=Local&parseTime=True.
-func initMysql(driverName string, initConnectionTimeout time.Duration) string {
-	mysqlConfig := client.CreateMySQLConfig(
-		common.GetStringConfigWithDefault(mysqlUser, "root"),
-		common.GetStringConfigWithDefault(mysqlPassword, ""),
-		common.GetStringConfigWithDefault(mysqlServiceHost, "mysql"),
-		common.GetStringConfigWithDefault(mysqlServicePort, "3306"),
-		"",
-		common.GetStringConfigWithDefault(mysqlGroupConcatMaxLen, "1024"),
-		common.GetMapConfig(mysqlExtraParams),
-	)
+// Initializes Database driver. Use `driverName` to indicate which type of DB to use:
+// 1) "mysql" for MySQL
+// 2) "pgx" for PostgreSQL
+func initDBDriver(driverName string, initConnectionTimeout time.Duration) string {
+	var sqlConfig, dbName string
+	var mysqlConfig *mysql.Config
+	switch driverName {
+	case "mysql":
+		mysqlConfig = client.CreateMySQLConfig(
+			common.GetStringConfigWithDefault(mysqlUser, "root"),
+			common.GetStringConfigWithDefault(mysqlPassword, ""),
+			common.GetStringConfigWithDefault(mysqlServiceHost, "mysql"),
+			common.GetStringConfigWithDefault(mysqlServicePort, "3306"),
+			"",
+			common.GetStringConfigWithDefault(mysqlGroupConcatMaxLen, "1024"),
+			common.GetMapConfig(mysqlExtraParams),
+		)
+		sqlConfig = mysqlConfig.FormatDSN()
+		dbName = common.GetStringConfig(mysqlDBName)
+	case "pgx":
+		sqlConfig = client.CreatePostgreSQLConfig(
+			common.GetStringConfigWithDefault(postgresUser, ""),
+			common.GetStringConfigWithDefault(postgresPassword, ""),
```

Comment:
Do we want to set a default user like mysql does?

Reply:
Yes, thanks for finding this! I should add a default user here.

Reply:
Added